indexer/services/vulcan/src/lib/on-message.ts

```diff
@@ -86,7 +99,40 @@ export async function onMessage(message: KafkaMessage): Promise<void> {
     const handler: Handler = new (getHandler(update))!(
       getTransactionHashFromHeaders(message.headers),
     );
-    await handler.handleUpdate(update);
+
+    // If headers don't exist, create them.
+    const headers = message.headers ?? {};
+    // If the message received timestamp doesn't exist
+    // (i.e when a short term order is directly sent to vulcan via full node)
+    // set the message_received_timestamp to the message timestamp and the event type
+    // to be a short term order event type.
+    if (!headers.message_received_timestamp) {
+      headers.message_received_timestamp = message.timestamp;
+    }
+    if (!headers.event_type) {
+      if (update.orderPlace) {
+        headers.event_type = 'ShortTermOrderPlacement';
+      } else if (update.orderRemove) {
+        headers.event_type = 'ShortTermOrderRemoval';
+      } else if (update.orderUpdate) {
+        headers.event_type = 'ShortTermOrderUpdate';
+      }
+    }
+
+    await handler.handleUpdate(update, headers);
+
+    const postProcessingTime: number = Date.now();
+    if (originalMessageTimestamp !== undefined) {
```

Comment:
Is this supposed to be using originalMessageTimestamp? Shouldn't we be grabbing the timestamp again due to what we're doing at L109 to L120?